docs/transfer_guard.rst

```diff
@@ -0,0 +1,70 @@
+Transfer guard
+==============
+
+JAX may transfer data between the host and devices and between devices during
+type conversion and input sharding. To log or disallow any unintended
+transfers, the user may configure a JAX transfer guard.
+
+JAX transfer guards distinguish between two types of transfers:
+
+* Explicit transfers: ``jax.device_put*()`` and ``jax.device_get()`` calls.
+* Implicit transfers: Other transfers (e.g., printing a ``DeviceArray``).
+
+A transfer guard can take an action based on its guard level:
+
+* ``"allow"``: Silently allow all transfers (default).
+* ``"log"``: Log and allow implicit transfers. Silently allow explicit
+  transfers.
+* ``"disallow"``: Disallow implicit transfers. Silently allow explicit
```

Comment:
IMO we should also make clear what does disallow means here, i.e. raise an error, and maybe also mention the error type.

Reply:
Thanks for the suggestion! I have updated the text.